src/sentry/quotas/base.py

```diff
@@ -322,6 +324,84 @@ def get_key_quota(self, key):
         limit, window = key.rate_limit
         return _limit_from_settings(limit), window
 
+    def get_project_abuse_quotas(self, org):
+        # Per-project abuse quotas for errors, transactions, attachments, sessions.
+        global_abuse_window = options.get("project-abuse-quota.window")
+
+        for option, compat_options, id, categories in (
+            (
+                "project-abuse-quota.error-limit",
+                (
+                    "sentry:project-error-limit",
+                    "getsentry.rate-limit.project-errors",
+                ),
+                "pae",
+                DataCategory.error_categories(),
+            ),
+            (
+                "project-abuse-quota.transaction-limit",
+                (
+                    "sentry:project-transaction-limit",
+                    "getsentry.rate-limit.project-transactions",
+                ),
+                "pat",
+                (DataCategory.TRANSACTION,),
+            ),
+            (
+                "project-abuse-quota.attachment-limit",
+                (),
+                "paa",
+                (DataCategory.ATTACHMENT,),
+            ),
+            (
+                "project-abuse-quota.session-limit",
+                (),
+                "pas",
+                (DataCategory.SESSION,),
+            ),
+        ):
+            limit = 0
+            abuse_window = global_abuse_window
+            # compat_options were previously present in getsentry
+            # for errors and transactions. The first one is the org
+            # option for overriding the global option, the second one.
+            # For now, these deprecated ones take precedence over the new
+            # to preserve existing behavior.
+            if compat_options:
+                limit = org.get_option(compat_options[0])
+                if not limit:
+                    limit = options.get(compat_options[1])
+
+            if not limit:
+                limit = org.get_option(option)
+                if not limit:
+                    limit = options.get(option)
+
+            if limit == 0:
+                # Unlimited.
+                continue
+
+            # Negative limits mean a reject-all quota, which
+            # cannot have an id.
+            if limit < 0:
+                limit = 0
```

Comment:
The contract in `QuotaConfig` is:

 - `None` means unlimited. The reason this exists is because we still count occurrences in the rate limiter, but for abuse purposes this can be completely omitted.
 - `0` means zero, reject everything.
 - positive number means limit.
 - negative number is invalid.

You can surely remap these semantics, and there's an existing function `_limit_from_settings` that converts this, but it doesn't support reject-all:

https://github.com/getsentry/sentry/blob/ae628557229334019c8339dd382ca1ec7e96a1d1/src/sentry/quotas/base.py#L179

If you'd like to support the full tri-state, it would be less error-prone to stick to the existing definition of (None, 0, positive) just to avoid mistakes.

Reply:
Confused about this. Surely we'd want to support reject-all? The only way to pass limit=0 to construct QuotaConfig is to set those options to a negative int to bypass the `not limit` branches.